tanner/emulators/template_injection.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+import logging
+
+from urllib.parse import unquote
+from tanner.utils import patterns
+from tanner.utils.aiodocker_helper import AIODockerHelper
+
+
+class TemplateInjection:
+
+    def __init__(self, loop=None):
+        self._loop = loop if loop is not None else asyncio.get_event_loop()
+        self.logger = logging.getLogger('tanner.template_injection')
+        self.docker_helper = AIODockerHelper()
+
+    async def get_injection_result(self, payload):
+        execute_result = None
+        github_remote_path = 'https://raw.githubusercontent.com/mushorg/tanner/master/docker/tanner/' \
+                             'template_injection/Dockerfile'
+
+        # Build the custom image
+        await self.docker_helper.setup_host_image(
+            remote_path=github_remote_path, tag='template_injection:latest')
+
+        if patterns.TEMPLATE_INJECTION_TORNADO.match(payload):
+
+            tornado_template = 'import tornado\n' \
+                               'from tornado.template import Template\n' \
+                               'code = "%s"\n' \
+                               'result = tornado.template.Template(code)\n' \
+                               'template_injection_result = result.generate()\n' \
+                               'print(template_injection_result)' % payload
+
+            cmd = ["python3", "-c", tornado_template]
+
+            execute_result = await self.docker_helper.execute_cmd(cmd, 'template_injection:latest')
+
+            # Removing string "b''" from results
+            if execute_result:
+                execute_result = execute_result[2:-2]
+
+        elif patterns.TEMPLATE_INJECTION_MAKO.match(payload):
+
+            mako_template = 'from mako.template import Template\n' \
+                            'mako_template = Template("""%s""")\n' \
+                            'template_injection_result = mako_template.render()\n' \
+                            'print(template_injection_result)' % payload
+
+            cmd = ["python3", "-c", mako_template]
```

Comment:
you duplicate the code, the `cmd` and `execute result` logic is similar for both templates

Reply:
Yes this I did this intentionally bcoz in tornado engine the execute result is having an extra string `"b''"` that I have to remove but in mako the output is ok.

But, I will try to remove redundancy more :)

Reply:
I think you can make changes in tornado/mako to return results in a standard way.

Reply:
Yes, im on it just need a little more time :)